test/mock_query_test.dart

```diff
@@ -278,6 +278,53 @@ void main() {
     }));
   });
 
+  test('Where queries on nested objects should return the correct snapshots', () async {
+    final instance = MockFirestoreInstance();
+    final bookmarks = await instance
+        .collection('users')
+        .document(uid)
+        .collection('bookmarks');
+    await bookmarks.add({
+      'hidden': false,
+      'pager': {'number': 10}
+    });
+    await bookmarks.add({
+      'tag': 'mostrecent',
+      'hidden': false,
+      'pager': {'number': 15}
+    });
+    await bookmarks.add({
+      'hidden': false,
+      'pager': {'number': 1}
+
+    });
+    await bookmarks.add({
+      'tag': 'mostrecent',
+      'hidden': true,
+      'pager': {'number': 3}
+    });
+    instance
+        .collection('users')
+        .document(uid)
+        .collection('bookmarks')
+        .where('pager.number', isEqualTo: 10)
+        .snapshots()
+        .listen(expectAsync1((QuerySnapshot snapshot) {
+      expect(snapshot.documents.length, equals(1));
+      expect(snapshot.documents.first.data['pager']['number'], equals(10));
+    }));
+
+    instance
+        .collection('users')
+        .document(uid)
+        .collection('bookmarks')
+        .where('pager.number', isGreaterThan: 9)
```

Comment:
Awesome tests! Could you add a final test that checks that for nonexistent paths, it does not throw an exception, but simply returns an empty list? Some thing like:

```dart
 instance
        .collection('users')
        .document(uid)
        .collection('bookmarks')
        .where('pager.some.deep.path', isEqualTo: 10)
        .snapshots()
        .listen(expectAsync1((QuerySnapshot snapshot) {
      expect(snapshot.documents.length, equals(0));
    }));
```